pages/posts/the-last-breaking-change.md

```diff
@@ -0,0 +1,78 @@
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "properties": {
+    "firstName": {
+      "type": "string",
+      "database-field-id": 1234
+    }
+  }
+}
+```
+
+a validator would either ignore `database-field-id` or include it and its value as an annotation in the output.  One can see how this feature would be extremely helpful.  However, this feature also blocks our ability to promise future compatibility.
+
+## Why can't we still support this?
+
+Suppose a user wrote the example schema above while using a validator written to conform to the 2023 version of the specification.  A couple years later, they want to upgrade the validator now that supports the 2025 specification.  However in that 2025 specification, we added `database-field-id` as a keyword, and its value is expected to be a string.  Suddenly the user's schema is no longer valid.  **We've broken a user by adding a new keyword.**
+
+In order to prevent this scenario, we are forced to forbid keywords that are not declared by a listed vocabulary.  We recognize that this will break a lot of people up front.  However we feel that a promise of future-compatible specifications should take precedence over the immediate pain of having to change your schemas yet again (and potentially again perpetually into the future).
+
+## What is being done to soften the blow?
+
+We have an [open discussion](https://github.com/orgs/json-schema-org/discussions/241) on this very topic, and we invite you to weigh in.
+
+Current proposals include things from simple prefixes on unknown keywords as a convention to indicate custom annotations to more complex solutions like inlined and ad-hoc vocabularies bundled into the schema that can define annotation-only keywords, like `title` and `readOnly`.  These options are effectively ways for the schema author to say, "I know these keywords aren't declared by any vocabulary, but I'd really like them in my schema.  Please disregard them."
+
+## Will there be any other breaking changes in the next release?
+
+Hopefully not.  We believe that removing support for unknown keywords will be the only change of any real negative impact, but that doesn't preclude the possibility of other changes that break things in less critical ways.
+
+There are generally two approaches to including a breaking change:
+
+- Find anything else that needs to change, breaking or not, and include it.  It's breaking anyway, so why not change what we can?
```

Comment:
To me this specific paragraph buries the lede a tiny bit -- as a user of JSON Schema I'd get a ton more worried hearing someone thinks the first thing here is a good idea, even though it's not necessarily relevant to the whole rest of the post which was essentially about convincing me about needing to change the behavior of unknown keywords.

I'd probably just drop it if it were me personally honestly (it being "There are generally two approaches" and the two bullets).

Reply:
Well, I do want to provide assurance that this isn't what we're going to do; that we've considered the various options and explicitly decided against this.

Reply:
I'm not sure I see where that assurance is, to me it read a bit like simply listing 2 options without deciding between them. But yeah leave it in obviously if you prefer it.

Reply:
If it's not clear, then I'll try to rework it.  I think the sentiment is important.

Reply:
I've updated it and just worked the second point into the following paragraph.  How does that read for you?

Reply:
Looks great!